tester/main.cpp

```diff
@@ -301,47 +300,41 @@ int main(int argc, char const *argv[]) {
 	// preparing versions
 	vc::Version::Builder builder;
 	std::list<run_element_t> jobList;
+	vc::vc_utils_init();
 
 	for (const auto &size : size_list) {
 		for (const auto &cat : configList) {
 			for (const auto &k : cat.kernel_config_list) {
 				for (const auto &datatype : k.data_type_option_list) {
 					const std::string label = cat.category_label + " " +
-					                          k.kernel_label + " - " +
-					                          size.first + " - " +
-					                          datatype.first;
-					builder._compiler = default_compiler;
+						k.kernel_label + " - " +
+						size.first + " - " +
+						datatype.first;
 					const std::string kernelSourceDir = polybench_source + "/" +
-					                                    cat.category_folder + "/" +
-					                                    k.kernel_folder;
-					vc::Option sourceDir("sourceDir", "-I", kernelSourceDir);
-					builder._optionList = {size.second, datatype.second, sourceDir};
-					builder._optionList.insert( builder._optionList.end(),
-					                            default_options.begin(),
-					                            default_options.end());
-					builder._fileName_src = kernelSourceDir + "/" + k.fileName;
-					builder._functionName = k.function_name;
-					jobList.push_back(std::make_pair(label, builder.build()));
+						cat.category_folder + "/" +
+						k.kernel_folder;
+					const vc::opt_list_t options = {
+						vc::make_option("-include polybench.c"),
+						vc::make_option("-include utils.hpp"),
+						vc::make_option("-O0"),
+						vc::make_option("-I" + polybench_source + "/utilities"),
+						vc::make_option("-I ../include"),
+						vc::make_option("-I" + kernelSourceDir),
+						size.second,
+						datatype.second,
+					};
+					vc::version_ptr_t version = vc::createVersion(kernelSourceDir + "/" + k.fileName, k.function_name, options);
+					kernel_t f = (kernel_t) vc::compileAndGetSymbol(version);
+					if (f) {
+						std::cout << "working on " << k.fileName << std::endl;
+						kernel_t to_run = (kernel_t) version->getSymbol(0);
+						auto v = std::vector<int>();
+						to_run(v);
+					}
 				}
 			}
 		}
 	}
 
-	// compiling and running
-	char* runningArgv[] = {};
-	for (const auto &j : jobList) {
-		std::cout << "working on " << j.first << std::endl;
-		bool ok = j.second->compile();
-		if (!ok) {
-			std::cerr << "Error while compiling " << j.first << std::endl;
-			continue;
-		}
-		main_signature_t* run = reinterpret_cast<main_signature_t*>(j.second->getSymbol());
-		if (run) {
-			run(0, runningArgv);
-		}
-		j.second->fold();
-	}
-
```

Comment:
I still prefer to keep the configuration loop separated from the compilation loop.

This way, it is more evident the separation of tasks and the capability of vc::Version to store the compilation state.